src/documentBuilder.ts

```diff
@@ -127,11 +131,20 @@ export class DocumentBuilder {
 
   /**
    * Add a single metadata key and value pair on the document. See {@link Document.metadata}
-   * @param key
-   * @param value
+   * @param {string} key
+   * @param {MetadataValue} value
+   * @param {Transformer} [metadataKeyTransformer=BuiltInTransformers.identity] The {@link Transformer} to apply to the metadata key.
+   * If not specified, no transformation will be applied to the metadata key.
+   *
+   * For a list of built-in transformers, use {@link BuiltInTransformers}.
    * @returns
    */
-  public withMetadataValue(key: string, value: MetadataValue) {
+  public withMetadataValue(
+    key: string,
+    value: MetadataValue,
+    keyTransformer = BuiltInTransformers.identity
```

Comment:
Prolly should type it :
```suggestion
    keyTransformer: Transformer = BuiltInTransformers.identity
```